### PR TITLE
[Dx-746] Add Tyk Plugin examples resources

### PR DESCRIPTION
**Pull request by dcs3spp**
[DX-746](https://tyktech.atlassian.net/browse/DX-746)

[preview](https://deploy-preview-3522--tyk-docs.netlify.app/docs/nightly/plugins/marketplace)

[DX-746]: https://tyktech.atlassian.net/browse/DX-746?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/plugins/marketplace.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Introduction
+
+Welcome to the Tyk Plugins Marketplace, dedicated to providing you with a curated list of resources that showcase how to develop Tyk Plugins. 
+
+[Tyk Plugins]({{< ref "plugins" >}}) are a powerful tool that allows you to develop custom middleware that can intercept requests at different stages of the request lifecycle, modifying/transforming headers and body content.
+
+Tyk has extensive support for writing custom plugins using a wide range of languages, most notably: Go, Python, Javascript etc. In fact, plugins can be developed using most languages via *gRPC*.
+
+## Blogs
```
Comment (JRWu) [inline tyk-docs/content/plugins/marketplace.md]: There are a couple more you could add to the list here:
https://tyk.io/blog/how-to-configure-a-grpc-server-using-tyk/
https://tyk.io/blog/how-to-deploy-python-plugins-in-tyk-running-on-kubernetes/

Or we could potentially list the blog url with the plugin search here:
https://tyk.io/?s=plugin

File: tyk-docs/content/plugins/marketplace.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+title: Tyk Plugins Marketplace
+description: "Contains links to repositories with example plugins"
+tags: ["custom", "plugin", "plugins marketplace", "marketplace", "go", "goplugins",  "go plugin", "tyk go plugin", "golang plugin", "Python", "Javascript", "JVMS"]
+---
+
+<!-- Want to try and get a design layout setup for this that uses stylesheets from home page to offer similar layout -->
+
+## Introduction
+
+Welcome to the Tyk Plugins Marketplace, dedicated to providing you with a curated list of resources that showcase how to develop Tyk Plugins.
```
Comment (JRWu) [inline tyk-docs/content/plugins/marketplace.md]: Is marketplace the proper terminology here? Perhaps I'm being pedantic but other vendors associate marketplace with a page where you can pick and choose different plugins as opposed to a set of urls

File: tyk-docs/content/plugins/marketplace.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+### 2. [Custom Plugin Examples](https://github.com/TykTechnologies/custom-plugin-examples/tree/master)
+- **Description**: Index of examples for a range of plugin hooks (Pre, Post, Post-Auth and Response) developed using a variety of languages.
+- **Key Features Demonstrated**: Specific examples include invoking an AWS lambda, inserting a new claim into a JWT, inject a signed JWT into authorization header, request header modification. A range of examples are available including Python, Java, Ruby, Javascript, NodeJS and Go.
+
+### 3. [Environment For Plugin Development](https://github.com/TykTechnologies/custom-go-plugin)
+- **Description**: Provides a docker-compose environment for developing your own custom plugins.
```
Comment (JRWu) [inline tyk-docs/content/plugins/marketplace.md]: Perhaps we need to be more explicit and state that this is only for development of custom Go plugins

**Review by JRWu — CHANGES_REQUESTED** [review]
Reads great, just requesting a couple minor changes and comments

File: tyk-docs/content/plugins/marketplace.md (hunk 2 above)
Comment (dcs3spp) [reply tyk-docs/content/plugins/marketplace.md]: Thanks @JRWu, how about Tyk Plugins Hub?

File: tyk-docs/content/plugins/marketplace.md (hunk 1 above)
Comment (dcs3spp) [reply tyk-docs/content/plugins/marketplace.md]: Nice!!! Thanks @JRWu for including the other blogs content and search URL. I have added both resources and linked to the plugin search also.

Comment (JRWu) [reply tyk-docs/content/plugins/marketplace.md]: Awesome, thank you

File: tyk-docs/content/plugins/marketplace.md (hunk 2 above)
Comment (JRWu) [reply tyk-docs/content/plugins/marketplace.md]: Love it

File: tyk-docs/content/plugins/plugin-hub.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- **Key Features Demonstrated**: Specific examples include invoking an AWS lambda, inserting a new claim into a JWT, inject a signed JWT into authorization header, request header modification. A range of examples are available including Python, Java, Ruby, Javascript, NodeJS and Go.
+
+#### 3. [Environment For Plugin Development](https://github.com/TykTechnologies/custom-go-plugin)
+- **Description**: Provides a docker-compose environment for developing your own custom Go plugins.
+- **Key Features Demonstrated**: Showcases support for bundling plugins, uploading plugins to AWS S3 storage, test coverage etc.
+
+## Conclusion
+
+[Tyk Plugins]({{< ref "plugins" >}}) is a valuable asset in the Tyk API Platform. Plugins enable you to develop custom middleware for intercepted upstream API requests and responses. The resources included in this Tyk Marketplace serve to help you harness the capabilities and use cases for Tyk Plugins. By exploring these resources and experimenting with the code, you can gain a deeper understanding of how to leverage [Tyk Plugins]({{< ref "plugins" >}}) to build innovative and efficient solutions for your projects.
```
Comment (JRWu) [inline tyk-docs/content/plugins/plugin-hub.md]: I would say "are a valuable asset to ..." in the first sentence

And then ultimately rename Marketplace to Plugin Hub in this sentence

**Review by JRWu — CHANGES_REQUESTED** [review]
Just two more minor changes requested please

File: tyk-docs/content/plugins/plugin-hub.md (hunk 4 above)
Comment (dcs3spp) [reply tyk-docs/content/plugins/plugin-hub.md]: Thanks @JRWu I have made the requested changes

**Review by JRWu — APPROVED** [review]
LGTM!

**Pull request merged**